core/Tracker/VisitExcluded.php

```diff
@@ -360,8 +360,18 @@ protected function isUserAgentExcluded()
 
         if (!empty($excludedAgents)) {
             foreach ($excludedAgents as $excludedUserAgent) {
-                // if the excluded user agent string part is in this visit's user agent, this visit should be excluded
-                if (stripos($this->userAgent, $excludedUserAgent) !== false) {
+                // Build up a regex if the string supplied is not a regular expression.
+                // The resulting regex is case-insensitive to match the behaviour of stripos().
+                $userAgentRegex = $excludedUserAgent;
+                if (preg_match('~^[^/].+?[^/]i?$~', $excludedUserAgent)) {
+                    $userAgentRegex = '/' . $userAgentRegex . '/i';
```

Comment:
Thanks for the review @sgiehl, I was trying to avoid a `@preg_match()` call with that trying to build a regex if supplied a plain string but on reflection it doesn't seem like a good idea. I could have used `preg_quote()` to escape special characters but it's probably more consistent to discard non-valid regular expressions and evaluate them with `stripos()` only as before.

I have had a look at the similar function you've highlighted and I don't think I can use it here without reworking it quite a bit. Instead, I've reworked the `isUserAgentExcluded()` method and updated the tests to include one for an invalid regex.

Reply:
Hi @sgiehl, is there anything else that needs to be done for this PR? I have noticed some UI tests are failing but it's unclear to me if that is anything to do with the changes proposed in this PR.